go-controller/pkg/util/multi_network.go

```diff
@@ -0,0 +1,338 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	ovncnitypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	kapi "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+var ErrorAttachDefNotOvnManaged = errors.New("net-attach-def not managed by OVN")
+
+// NetInfo is interface which holds network name information
+// for default network, this is set to nil
+type NetInfo interface {
+	GetNetworkName() string
+	IsSecondary() bool
+	GetPrefix() string
+	AddNAD(nadName string)
+	DeleteNAD(nadName string)
+	HasNAD(nadName string) bool
+}
+
+type DefaultNetInfo struct{}
+
+// GetNetworkName returns the network name
+func (nInfo *DefaultNetInfo) GetNetworkName() string {
+	return types.DefaultNetworkName
+}
+
+// IsSecondary returns if this network is secondary
+func (nInfo *DefaultNetInfo) IsSecondary() bool {
+	return false
+}
+
+// GetPrefix returns if the logical entities prefix for this network
+func (nInfo *DefaultNetInfo) GetPrefix() string {
+	return ""
+}
+
+// AddNAD adds the specified NAD, no op for default network
+func (nInfo *DefaultNetInfo) AddNAD(nadName string) {
+}
+
+// DeleteNAD deletes the specified NAD, no op for default network
+func (nInfo *DefaultNetInfo) DeleteNAD(nadName string) {
+}
+
+// HasNAD returns true if the given NAD exists, already return true for
+// default network
+func (nInfo *DefaultNetInfo) HasNAD(nadName string) bool {
+	// default network always needs to be plumbed over Pod
+	return true
+}
```

Comment:
I would panic in these methods. In the case of `HasNAD` if it is not called with `ovntypes.DefaultNetworkName` as NAD name maybe.